analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,219 @@
+  table <- broom::tidy(fit) 
+  
+  # Save the table data in a TSV
+  readr::write_tsv(table, file.path(results_dir, paste0("cox_reg_results_per_", ind_var, ".tsv")))
+  
+  print(table)
+  
+  # output the plot
+  pdf(file.path(plots_dir, paste0("coxph_survival_per_", ind_var, ".pdf")))
+  plot_coxph <- survminer::ggsurvplot(survfit(fit),
+                                      data=meta_indep,
+                                      xlim = c(0, 5000),
+                                      break.time.by = 500,
+                                      pval = TRUE, 
+                                      conf.int = TRUE,
+                                      risk.table = TRUE, # Add risk table
+                                      linetype = "strata", # Change line type by groups
+                                      surv.median.line = "hv", # Specify median survival
+                                      ggtheme = theme_bw())
+  print(ggpar(plot_coxph[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_coxph[[2]], font.legend=6))
+
+  dev.off()
+
+  print(ggpar(plot_coxph[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_coxph[[2]], font.legend=6))
+}
+
+```
+
+## Log Rank analysis 
+### Generate output for categorical files 
+```{r}
+ind_var_list <- c("hgg_group", "cancer_group_over10")
+
+for(i in 1:length(ind_var_list)){
+  ind_var <- ind_var_list[i]
+
+  # define model
+  model <- paste0("survival::Surv(time = OS_days, event = OS_status_fixed) ~ ", ind_var)
+  
+  # run survival analysis
+  fit <- survival::survdiff(formula(model),
+                            data = meta_indep)
+  # Obtain p value for Chi-Squared stat
+  fit$p.value <- pchisq(fit$chisq, df = length(fit$n) - 1, lower = FALSE)
+  
+  print(fit)
+  # save the output
+  saveRDS(fit, file.path(results_dir, paste0("log_rank_survival_per_", ind_var, ".RDS")))
+  
+  # generate plots fit
+  fit_plot <- survfit(formula(model), data = meta_indep)
+  
+  # output the plot
+  pdf(file.path(plots_dir, paste0("logrank_survival_per_", ind_var, ".pdf")))
+  plot_logrank <- survminer::ggsurvplot(fit_plot,
+                                        data=meta_indep,
+                                        xlim = c(0, 5000),
+                                        break.time.by = 500,
+                                        pval = TRUE, 
+                                        conf.int = TRUE,
+                                        risk.table = TRUE, # Add risk table
+                                        linetype = "strata", # Change line type by groups
+                                        surv.median.line = "hv", # Specify median survival
+                                        ggtheme = theme_bw())
+  print(ggpar(plot_logrank[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_logrank[[2]], font.legend=6))
+
+  dev.off()
+
+  print(ggpar(plot_logrank[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_logrank[[2]], font.legend=6))
+}
+
+```
+
```

Comment:
It will be nice to keep the univariate analysis above, with the the potential modification of also testing broad histology as an additional univariate predictor. Here is where we can dive into the multivariate analysis, specifically with `tp53_score + telomerase_score + hgg_group` as the model parameters, or maybe tp53_score + telomerase_score + broad_histology` as an additional thing to try.

Once that model is fit, I like `survminer::ggforest()` as a solid visualization tool for Cox PH models.

As a reader, I would also be interested to know what the bivariate distribution of tp53_score and telomerase_score looks like, stratified by HGAT status. A 2D density plot with different colors for HGAT status comes to mind. 🤔

Reply:
@envest, I have now ran multivariate analysis for `tp53_score + telomerase_score + hgg_group` as well as `tp53_score + telomerase_score + broad_histology` and add `ggforest` for display. It looks like that second analysis had some issues with converging and the figure is not showing - advice on how to solve that will be appreciated.

Reply:
Nice! The `hgg_group` model looks good to me. For `broad_histology`, the idea was to try to get fewer, larger groups in the model than we had with `cancer_group`. The model not converging means the groups are still too numerous and/or small such that we just don't have sufficient data in all parts of the model space. This could be a decision point -- either keep dropping small groups from the model until the model converges, or just focus on the HGAT/nonHGAT analysis.

Reply:
One thought I had looking at the forest plot... my preference would be to recode non-HGAT as the reference group to be able to see the change in hazard ratio for HGAT.

Reply:
Thanks for the input - I have now changed non-HGAT as the reference group and re-ran the analysis. 
Additionally, since the broad histology is not converting - I tried using another column called `oncoprint_group` - this categorizes all cancers into 4 groups and the results are included. Seemed to be informative as well! Also added a density plots stratified by these groups.